pkg/ruler/notifier.go

```diff
@@ -74,68 +75,86 @@ func (rn *rulerNotifier) stop() {
 // Builds a Prometheus config.Config from a ruler.Config with just the required
 // options to configure notifications to Alertmanager.
 func buildNotifierConfig(rulerConfig *Config) (*config.Config, error) {
-	if rulerConfig.AlertmanagerURL.URL == nil {
-		return &config.Config{}, nil
-	}
+	validURLs := make([]*url.URL, 0, len(rulerConfig.AlertmanagerURL))
 
-	u := rulerConfig.AlertmanagerURL
-	var sdConfig sd_config.ServiceDiscoveryConfig
-	if rulerConfig.AlertmanagerDiscovery {
-		if !strings.Contains(u.Host, "_tcp.") {
-			return nil, fmt.Errorf("When alertmanager-discovery is on, host name must be of the form _portname._tcp.service.fqdn (is %q)", u.Host)
+	srvDNSregexp := regexp.MustCompile(`^_.+._.+`)
+	for _, h := range rulerConfig.AlertmanagerURL {
+		url, err := url.Parse(h)
+		if err != nil {
+			return nil, err
 		}
-		dnsSDConfig := dns.SDConfig{
-			Names:           []string{u.Host},
-			RefreshInterval: model.Duration(rulerConfig.AlertmanagerRefreshInterval),
-			Type:            "SRV",
-			Port:            0, // Ignored, because of SRV.
-		}
-		sdConfig = sd_config.ServiceDiscoveryConfig{
-			DNSSDConfigs: []*dns.SDConfig{&dnsSDConfig},
+
+		if url.String() == "" {
+			continue
 		}
-	} else {
-		sdConfig = sd_config.ServiceDiscoveryConfig{
-			StaticConfigs: []*targetgroup.Group{
-				{
-					Targets: []model.LabelSet{
-						{
-							model.AddressLabel: model.LabelValue(u.Host),
-						},
-					},
-				},
-			},
+
+		// Given we only support SRV lookups as part of service discovery, we need to ensure
+		// hosts provided follow this specification: _service._proto.name
```

Comment:
`_portname._proto.name`

Reply:
According to [wikipedia](https://en.wikipedia.org/wiki/SRV_record) this is suppose to be `_service`, I believe the port name convention `_portname` is a k8s thing.

Reply:
I think Josh is right.